x/amm/client/cli/tx_swap_by_denom.go

```diff
@@ -20,17 +20,18 @@ const (
 
 func CmdSwapByDenom() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:     "swap-by-denom [amount] [denom-in] [denom-out]",
+		Use:     "swap-by-denom [amount] [denom-in] [denom-out] [recipient]",
 		Short:   "Swap an exact amount of tokens for a minimum of another token or a maximum amount of tokens for an exact amount on another token, similar to swapping a token on the trade screen GUI.",
 		Example: "elysd tx amm swap-by-denom 1000000000uatom uatom uusd --min-amount=1000000000uatom --max-amount=1000000000uatom --discount=0.1 --from jack --keyring-backend test",
-		Args:    cobra.ExactArgs(3),
+		Args:    cobra.ExactArgs(4),
 		RunE: func(cmd *cobra.Command, args []string) (err error) {
 			argAmount, err := sdk.ParseCoinNormalized(args[0])
 			if err != nil {
 				return err
 			}
 			argDenomIn := args[1]
 			argDenomOut := args[2]
+			argRecipient := args[3]
```

Comment:
@jelysn can you make the recipient a flag instead of a fixed command argument so that it is optional, thanks